ui/components/component-library/button-icon/button-icon.tsx

```diff
@@ -11,10 +11,15 @@ import {
 } from '../../../helpers/constants/design-system';
 
 import Box from '../../ui/box';
-import { Icon } from '../icon';
+import { Icon, IconSize } from '../icon';
 
 import { ButtonIconSize, ButtonIconProps } from './button-icon.types';
 
+const buttonIconSizeToIconSize: Record<ButtonIconSize, IconSize> = {
```

Comment:
ChatGPT: We use Record here to define a type that describes a mapping between ButtonIconSize and IconSize enum values.

The Record type is a built-in TypeScript utility type that creates a new type with a set of properties defined by the provided keys, each with a value of the same type. In this case, the keys are the ButtonIconSize enum values, and the values are the corresponding IconSize enum values.

Using Record provides a way to ensure that each ButtonIconSize value is mapped to a corresponding IconSize value. If there is a missing or incorrect mapping, TypeScript will raise a type error.

In the specific case of buttonIconSizeToIconSize, we are mapping the ButtonIconSize values of Sm and Lg to their corresponding IconSize values of Sm and Lg respectively.